shared/js/initApp.js

```diff
@@ -118,9 +119,42 @@ window.apps = {
       }
     })(appOptions.level);
   },
+
+  // Set up projects, skipping blockly-specific steps. Designed for use
+  // by levels of type "external".
+  setupProjectsExternal: function() {
+    dashboard.project = project;
```

Comment:
Nit: I'd rather see this function contain https://github.com/code-dot-org/code-dot-org/blob/b92425a5fdff00fe5ff49bde0f70a7d1d1803ee1/shared/js/initApp.js#L19-L22 and replace those lines with a call to this function.